lib/segment/src/id_tracker/id_tracker_base.rs

```diff
@@ -144,3 +145,146 @@ pub trait IdTracker {
 }
 
 pub type IdTrackerSS = dyn IdTracker + Sync + Send;
+
+pub enum IdTrackerEnum {
+    SimpleIdTracker(SimpleIdTracker),
+    ImmutableIdTracker(SimpleIdTracker),
```

Comment:
We might want to call the variants `Mutable` and `Immutable` for the time being.